### ISSUE TITLE
Minor version is not bumped when using non-default patterns

### ISSUE THREAD

**Issue by serpro69** [PaulHatch/semantic-version#153]
This is the output of the action with debug:

```
with:
    tag_prefix: v
    major_pattern: [major]
    minor_pattern: [minor]
    version_format: ${major}.${minor}.${patch}-prerelease${increment}
    bump_each_commit: false
    search_commit_body: true
    enable_prerelease_mode: true
    debug: true
    version_from_branch: false
    branch: HEAD
    use_branches: false
    user_format_type: csv

Version is 0.0.5-prerelease0
To create a release for this version, go to ...
```

The last tag in the repo is `v0.0.4`.

The commit message is as follows:

```
Uses maven ci-friendly-versions for versioning the project, instead of
using a hardcoded version.

Add build and pr workflows

Release [minor]
```

I thought this should have bumped next minor version. Am I misunderstanding some configuration bits or how this action is supposed to work?

Also a bit confusing why the next version contains `prerelease0` string. Is that because the action didn't calculate the next version and defaulted to patch bump with pre-release string?

**Comment by serpro69** [comment]
OK, I figured out it was because `enable_prerelease_mode: true`, which uses major for minor, and minor for patch... This config parameter is quite confusing, so I disabled it and now it works as expected.
However, I'm still getting `Version is 0.1.0-prerelease0` in the output. Why is it set to `prerelease`?

**Comment by oliveira-flavio-br** [comment]
> OK, I figured out it was because `enable_prerelease_mode: true`, which uses major for minor, and minor for patch... This config parameter is quite confusing, so I disabled it and now it works as expected. However, I'm still getting `Version is 0.1.0-prerelease0` in the output. Why is it set to `prerelease`?

I think you're getting the prerelease because you set that on version_format. Try removing the -prerelease${increment} from it.

**Comment by serpro69** [comment]
Thanks for the comment @oliveira-flavio-br . Yeah, it was a bit confusing at first. I do get that version format has pre-release in it, just didn't quite understand what role pre-release plays in this action and why it's always created even when I "say": "give me a version with next minor bump", not "give me a pre-release". From what I understand now, it's just an increment of the number of commits from previous version.

It's even more confusing because the task says:

```
Run paulhatch/semantic-version@v5.4.0
Version is 0.1.0-rc.1
```

But the output of the task actually contains a proper version tag w/o `rc` postfix:

```json
{
    "version": "0.1.0-rc.1",
    "major": "0",
    "minor": "1",
    "patch": "0",
    "increment": "1",
    "version_type": "minor",
    "changed": "true",
    "is_tagged": "false",
    "version_tag": "v0.1.0",
    "authors": "Serhii P <22973227+serpro69@users.noreply.github.com>",
    "previous_commit": "8a6c826a3056b4f14e92be8439d0656e0d69059a",
    "previous_version": "0.0.4",
    "current_commit": "c4b181053f3f660c694e34cccc11e1ac0138ad6d",
```

At least the way I've used pre-releases before with other tools, and in my own gradle semantic-versioning plugin, it doesn't make sense to create a new pre-release on each commit, because pre-release is also tagged.

In any case, I'm going to close this since I figured it out and made things work, just forgot to add a comment here.

Thanks again
